fabtests/common/shared.c

```diff
@@ -526,36 +539,42 @@ int ft_alloc_msgs(void)
 		ft_set_tx_rx_sizes(&tx_mr_size, &rx_mr_size);
 		rx_size = FT_MAX_CTRL_MSG + ft_rx_prefix_size();
 		tx_size = FT_MAX_CTRL_MSG + ft_tx_prefix_size();
-		buf_size = rx_size + tx_size;
+		half_buf_size = MAX(tx_size, rx_size);
 	} else {
 		ft_set_tx_rx_sizes(&tx_size, &rx_size);
 		tx_mr_size = 0;
 		rx_mr_size = 0;
-		buf_size = MAX(tx_size, FT_MAX_CTRL_MSG) * opts.window_size +
-			   MAX(rx_size, FT_MAX_CTRL_MSG) * opts.window_size;
+		half_buf_size = MAX(MAX(rx_size, FT_MAX_CTRL_MSG),
+				    tx_size) * opts.window_size;
 	}
 
 	/* Allow enough space for RMA to operate in a distinct memory
 	 * region that ft_sync() won't touch.
 	 */
 	rma_resv_bytes = FT_RMA_SYNC_MSG_BYTES +
 			 MAX( ft_tx_prefix_size(), ft_rx_prefix_size() );
-	buf_size += rma_resv_bytes*2;
+	half_buf_size += rma_resv_bytes;
 
 	if (opts.options & FT_OPT_ALIGN && !(opts.options & FT_OPT_USE_DEVICE)) {
 		alignment = sysconf(_SC_PAGESIZE);
 		if (alignment < 0)
 			return -errno;
-		buf_size += alignment;
-
+	}
+	half_buf_size = ft_get_aligned_size(half_buf_size, alignment);
+	buf_size = 2 * half_buf_size;
+	if (!(opts.options & FT_OPT_USE_DEVICE)) {
 		ret = posix_memalign((void **) &buf, (size_t) alignment,
 				buf_size);
 		if (ret) {
 			FT_PRINTERR("posix_memalign", ret);
 			return ret;
 		}
 	} else {
-		ret = ft_hmem_alloc(opts.iface, opts.device, (void **) &buf, buf_size);
+		/* allocate extra "alignment" bytes, to handle the case
+		 * "buf" is not aligned.
+		 */
+		ret = ft_hmem_alloc(opts.iface, opts.device, (void **) &buf,
+				    buf_size + alignment);
```

Comment:
It's confusing that buf_size isn't actually the size of the buffer.

Reply:
The comment was trying to explain the reason, does that help?

Reply:
I get the comment and why the adjustment is made.  It's that the variable named 'buf_size' isn't actually the size of the buffer.  I'd modify to:
```
buf_size += alignment;
ret = ft_hmem_alloc(..., buf_size);
```
So it's clear that buf_size is always the size of the buffer.

Reply:
I see. will do that

Reply:
@shefty I changed to `buf_size += alignment` as you suggested.

Along with this change, I had to desert the variable `half_buf_size` because with the added `alignemnt`, `half_buf_size` is not 1/2 of `buf_size`. I replaced it with `tx_buf_size` and `rx_buf_size`.